### ISSUE TITLE
Underscore in Middle of Variable Name

### ISSUE THREAD

**Issue by rbirchtree** [palantir/tslint#4697]
### Bug Report

- __TSLint version__: ^5.7.0"
- __TypeScript version__: "3.16
- __Running TSLint via__: (pick one) Studio 

#### TypeScript code being linted
This is for material-ui's library.
```ts

var total = this.props.steps.length;
    var li_width = 100 / total;
    var total_steps = this.props.steps.length;
    
```

with `tslint.json` configuration:

```json
{
  "compilerOptions": {
    "checkJs": false
  },
  "extends": ["tslint:recommended", "tslint-react", "tslint-config-prettier"],
  "rules" : {
    "no-console" : false, // disable for TS
    "object-literal-sort-keys": false,
    "ordered-imports" : false,
    "no-unused-expression": [true, "allow-fast-null-checks"],
    // allow namespaces to organize interface definitions in TypeDoc
    "no-namespace": false,
    "no-string-literal": false,
    //added to object access via string literals is allowed
    // no shadowing allowed except for types and interfaces so we can have same interface or type in different
    // namespaces
    "variable-name": [
      true,
      "ban-keywords",
      "check-format",
      "allow-leading-underscore", 
      "allow-pascal-case"
    ],
    //added to compile jsx variables;
    "no-shadowed-variable": [
      true,
      {
        //"class": false,
        //"enum": false,
        //"function": false,
        "interface": false,
        //"namespace": false,
        "typeAlias": false
        //"typeParameter": false,
        //"import": false
      }
    ]
  },
  "jsRules": {
    "no-console" : false, // disable for JS
    "object-literal-sort-keys": false,
    "no-shadowed-variable": false,
    "no-string-literal":false, 
    "variable-name": [
      true,
      "ban-keywords",
      "check-format",
      "allow-leading-underscore", 
      "allow-pascal-case"
    ]
  },
  "linterOptions": {
    "exclude": [
      "config/**/*.js",
      "node_modules/**/*.ts",
      "coverage/lcov-report/*.js"
    ]
  }
}

```

#### Actual behavior
(179,9): variable name must be in lowerCamelCase or UPPER_CASE

#### Expected behavior
No underscore in variable name....or passed the variable as is with underscore.

**Comment by adidahiya** [comment]
you want the `"allow-snake-case"` option. see the rule config documentation: https://palantir.github.io/tslint/rules/variable-name/
